modules/openthread/platform/crypto_psa.c

```diff
@@ -104,6 +104,16 @@ static otError getKeyRef(otCryptoKeyRef *aInputKeyRef, psa_key_attributes_t *aAt
 		if (psa_get_key_algorithm(aAttributes) == 0) {
 			psa_set_key_algorithm(aAttributes, PSA_ALG_HMAC(PSA_ALG_SHA_256));
 		}
+
+		/* KMU does not support deterministic ECDSA, so we need to set it to
+		 * PSA_ALG_ECDSA(PSA_ALG_SHA_256).
+		 * To keep backward compatibility with the previous functionality we must
+		 * leave PSA_ALG_DETERMINISTIC_ECDSA(PSA_ALG_SHA_256) for the ITS purposes.
+		 */
+		if (psa_get_key_algorithm(aAttributes) ==
+		    PSA_ALG_DETERMINISTIC_ECDSA(PSA_ALG_SHA_256)) {
+			psa_set_key_algorithm(aAttributes, PSA_ALG_ECDSA(PSA_ALG_SHA_256));
```

Comment:
Isn't a dead code? For KMU we do always set non-deterministic ECDSA right? And with ITS we will not enter here.

Reply:
OR.. it's due to this: https://github.com/nrfconnect/sdk-nrf/pull/21318/files#diff-2bd5a0038ec03774612f48f90657e0c19bbdea78a9c85b35c8058c40ea798b5eR763

Reply:
Discussed F2F.